### ISSUE TITLE
does --num-threads work? 

### ISSUE THREAD

**Issue by grenaud** [Illumina/DRAGMAP#29]
I tried specifying --num-threads but it seems that it takes all available threads by default, it there any way to change this behavior?

**Comment by rpetrovski** [comment]
It limits the number of threads used for compute-heavy processing (mapping, smith-waterman and such). Depending on workflow, there is roughly twice the --num-threads allocated.

**Comment by grenaud** [comment]
I would like this issue reopened, even when I specify the --num-threads to say 4, it still takes all of the cores available.

**Comment by grenaud** [comment]
ok I left it to run, it maxes out to the maximum CPU and then writes "finished decompress" then goes down to 4 cores. Any chance it could be capped to 4 cores for the phase before "finished decompress"

**Comment by rpetrovski** [comment]
Indeed, hash table decompression uses all available threads. This might take some time to get sorted. Meanwhile you can run 
`dragen-os --ht-reference ./ --ht-uncompress true`
inside the reference folder to avoid this being done for every run.
Afterwards be sure to specify --ref-load-has-bin true when you run dragen-os.

For quick runs with a handful or reads it might be beneficial to use `--mmap-reference true`

**Comment by dpryan79** [comment]
This also happens during construction of the index. That step seems to be rather explicit about ignoring the --num-threads option. For example I specified 4 but it's reporting `Spawning 12 threads to hash reference seeds for 2 passes`.

**Update**: apparently it's a different parameter (--ht-num-threads). I'd suggest changing that, it's incredibly non-standard.
